src/components/v2v3/shared/FundingCycleConfigurationDrawers/NftDrawer/index.tsx

```diff
@@ -53,22 +61,51 @@ export default function NftDrawer({
     theme,
     theme: { colors },
   } = useContext(ThemeContext)
+
+  const {
+    nftRewards: { rewardTiers: initialRewardTiers },
+  } = useContext(NftRewardsContext)
+
   const dispatch = useAppDispatch()
   const {
-    nftRewards: { rewardTiers: savedRewardTiers },
+    nftRewards,
     projectMetadata: { name: projectName, logoUri, infoUri },
   } = useAppSelector(state => state.editingV2Project)
 
+  const reduxRewardTiers = useMemo(() => nftRewards.rewardTiers, [nftRewards])
+
+  const { signer } = useWallet()
+
+  const { fundingCycleMetadata } = useContext(V2V3ProjectContext)
+
   const [marketplaceForm] = useForm<MarketplaceFormFields>()
   const [postPayModalForm] = useForm<NftPostPayModalFormFields>()
 
   const [addTierModalVisible, setAddTierModalVisible] = useState<boolean>(false)
   const [submitLoading, setSubmitLoading] = useState<boolean>(false)
 
   const [rewardTiers, setRewardTiers] = useState<NftRewardTier[]>(
-    savedRewardTiers ?? [],
+    reduxRewardTiers ?? [],
   )
 
+  useEffect(() => {
+    if (!rewardTiers.length) {
+      setRewardTiers(reduxRewardTiers ?? [])
+    }
+  }, [reduxRewardTiers, rewardTiers])
+
+  // a list of the `tierRanks` (IDs) of tiers that have been edited
+  const [editedRewardTierIds, setEditedRewardTierIds] = useState<number[]>([])
+
+  const editingRewardTierIDsPush = (tierId: number | undefined) => {
```

Comment:
**question** this feels fragile. Is there a solution where we could compute the edited tier IDs based on `rewardTiers` and `reduxRewardTiers` maybe?

Reply:
I don't see how this wouldn't work - every time a tier is edited, if it has an ID (already been launched), add its ID to `editedRewardTierIds` as long as it isn't already in there.

I think comparing `rewardTiers` and `reduxRewardTiers` would be quite inefficient. We'd have to filter `rewardTiers` for all tiers that (1) have an ID and (2), the corresponding tier for that ID in `reduxRewardTiers` is different. Since we can't do quick lookups by IDs as is, that would be kinda nasty.